tools/deploy-static.sh

```diff
@@ -1,23 +1,37 @@
 #!/usr/bin/env bash
-# Script to prepare Docs for Cloudflare Pages deployment 
+
+# Script to prepare Docs for Cloudflare Pages deployment
 
 set -euo pipefail
 
-dirname=$(dirname $0)
+# resolve absolute path and account for spaces, etc
+dirname="$(cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P)"
 cd "${dirname}"/.. || exit 1
 
 # Build folder is uploaded to Cloudflare
 cp -r static/* build/
 
-# Extract only the Cloudflare Redirects from redirect.txt
-DIVISION_LINE=`awk '/####*/ {print NR}' config/redirects.txt | sed -n 3p`
-head -n $DIVISION_LINE config/redirects.txt > build/_redirects
+# normalize all Cloudflare Style Redirects to include both trailing slash and no trailing slash variants
+generated="$(sed -rn 'p; s/^(\/([^ ]+\/)?[^\/ ]+) \//\1\/ \//p' config/redirects.txt |
+    sed -rn 'p; s/^(\/([^ ]+\/)?[^\/ ]+)\/ \//\1 \//p' |
+    awk '!/^(\/([^ ]+\/)?[^\/ ]+)/ || !seen[$0]++' | sort)"
+current="$(sort config/redirects.txt)"
+
+if [ "${current}" != "${generated}" ]
+then
+    echo "Error: one or more entries config/redirects.txt is missing the trailing-slash or non-trailing slash variant, both must exist"
```

Comment:
This is making things harder than they should be I feel. I think we should have this validation occurring for the file that is being created on build time which is build/_redirects (output)

Which is actually the file being sent to Cloudflare.

Reply:
Made the change to validate the build output rather than the input. The impact should be the same.